### PR TITLE
Fix resetting of paper-input and paper-textarea

### PR DESCRIPTION
**Pull request by notwaldorf**
Fixes https://github.com/PolymerElements/iron-form/issues/27.

### CODE REVIEW

File: test/basic.html [hunk 1]
```diff
@@ -16,6 +16,8 @@
   <script src="../../web-component-tester/browser.js"></script>
 
   <link rel="import" href="../../polymer/polymer.html">
+  <link rel="import" href="../../paper-input/paper-input.html">
+  <link rel="import" href="../../paper-input/paper-textarea.html">
```
Comment (valdrinkoshi) [inline test/basic.html]: You should import `paper-textarea` in the `bower`'s `devDependencies`

File: test/basic.html [hunk 2]
```diff
@@ -376,11 +390,50 @@
...
+        assert.equal(paperInput.inputElement.value, 'zug');
+        assert.equal(paperTextarea.value, 'zog');
+        assert.equal(paperTextarea.inputElement.textarea.value, 'zog');
         done();
       });
 
       form.reset();
+    });
+
+    test('form restores the default values if no changes were made', function(done) {
```
Comment (valdrinkoshi) [inline test/basic.html]: this seems a duplicate of the previous test

File: test/basic.html [hunk 3]
```diff
@@ -376,11 +390,50 @@
         assert.equal(input.value, 'bar');
         assert.isTrue(checkbox1.checked);
         assert.isFalse(checkbox2.checked);
+        assert.equal(paperInput.value, 'zug');
+        assert.equal(paperInput.inputElement.value, 'zug');
+        assert.equal(paperTextarea.value, 'zog');
+        assert.equal(paperTextarea.inputElement.textarea.value, 'zog');
```
Comment (valdrinkoshi) [inline test/basic.html]: back to the origins! :beers:

File: test/basic.html (hunk 1 above)
Comment (notwaldorf) [inline test/basic.html]: It's in the same package (`paper-input`), which is already a dependency

File: test/basic.html (hunk 2 above)
Comment (notwaldorf) [inline test/basic.html]: Nope, the previous tests takes inputs, makes changes to them, then restores them. This test takes inputs and restores them immediately. You'd be surprised how one failed and the other didn't :)

Comment (valdrinkoshi) [inline test/basic.html]: oh true, so subtle! Good test! :+1:

File: test/basic.html (hunk 1 above)
Comment (valdrinkoshi) [inline test/basic.html]: word.

**Comment by valdrinkoshi** [comment]
LGTM!

**Comment by dman777** [comment]
I am not sure if it is this PR, but there is breakage where form reset causes breakage in paper-radio buttons and paper-dropdown-menu. After a reset, the default selections will no longer show as they did when a form first loads.

**Pull request merged**